relayer/src/keyring.rs

```diff
@@ -343,15 +345,33 @@ impl KeyRing {
     }
 
     /// Sign a message
-    pub fn sign_msg(&self, key_name: &str, msg: Vec<u8>) -> Result<Vec<u8>, Error> {
+    pub fn sign_msg(
+        &self,
+        key_name: &str,
+        msg: Vec<u8>,
+        address_type: &AddressType,
+    ) -> Result<Vec<u8>, Error> {
         let key = self.get_key(key_name)?;
 
         let private_key_bytes = key.private_key.private_key.to_bytes();
-        let signing_key =
-            SigningKey::from_bytes(private_key_bytes.as_slice()).map_err(Error::invalid_key)?;
-
-        let signature: Signature = signing_key.sign(&msg);
-        Ok(signature.as_ref().to_vec())
+        match address_type {
+            AddressType::Cosmos => {
+                let signing_key = SigningKey::from_bytes(private_key_bytes.as_slice())
+                    .map_err(Error::invalid_key)?;
+                let signature: Signature = signing_key.sign(&msg);
+                Ok(signature.as_ref().to_vec())
+            }
+            AddressType::Ethermint { .. } => {
+                let hash = keccak256_hash(msg.as_slice());
+                let s = Secp256k1::signing_only();
+                // SAFETY: hash is 32 bytes, as expected in `Message::from_slice` -- see `keccak256_hash`, hence `unwrap`
+                let sign_msg = Message::from_slice(hash.as_slice()).unwrap();
+                let key = SecretKey::from_slice(private_key_bytes.as_slice())
+                    .map_err(Error::invalid_key_raw)?;
+                let (_, sig_bytes) = s.sign_recoverable(&sign_msg, &key).serialize_compact();
+                Ok(sig_bytes.to_vec())
+            }
+        }
```

Comment:
I think this is still required because I am still seeing the same error without it.
To reproduce the error, start two Ethermint chains with keys added using `--algo=secp256k1` and run `tx raw create-client ethermint_901-1 ethermint_902-2`. 
```
$ hermes -c tx raw create-client ethermint_901-1 ethermint_902-2
Success: ChainError(
    "deliver_tx on chain ethermint_901-1 for Tx hash E95AE4451496F4E360193D384FFCC75CFCF1763B5F2C443ED550BCAFED5E7794 reports error: code=Err(4), log=Log(\"signature verification failed; please verify account number (0) and chain-id (ethermint_901-1): unauthorized\")",
)

```

Reply:
ok, added the check on `sign_msg`